### PR TITLE
695 - blog template frontend

### PR DESCRIPTION
**Pull request by omonk**
## Description - [Trello ticket](https://trello.com/c/sburipjF/695-create-blog-post-template)

Addresses the frontend work required for blog work

- Adds MDX plugins to gatsby
- Creates MDX nodes for graphql in gatsby-node
- Creates typography styles for blog
- Adds stories for all components used within the blog 
- Adds SEO meta content to head for each blog post
- Move blog page to templates
- Adds paging to blog template 

## Checklist

- [x] Updating or creating a new page? Consider any SEO requirements such as:
  - [x] Does the page have an `h1` tag?
  - [x] Does the page have a correctly formed meta title?
  - [x] Does the page have a correctly formed meta description?
  - [x] Does the page have a unique `og:image` tag (if required)?
- [ ] checked that this PR resolves the spec provided from trello / this description

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -173,6 +186,82 @@ exports.createPages = async ({ graphql, actions }) => {
...
+   */
+
+  const allBlogPosts = result.data.allContentfulBlogPost.edges
+  const postsPerPage = 6
+
+  const numberOfPages = Math.ceil(allBlogPosts.length / postsPerPage)
+
+  Array.from({
+    length: numberOfPages
+  }).forEach((_, i) => {
```
Comment (jeysal) [inline gatsby-node.js]: `Array(numberOfPages).forEach`

File: src/components/Blog/Typography.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    line-height: ${remcalc(33)};
+  `}
+
+  font-weight: 400;
+  color: ${props => props.theme.colors.text};
+  padding: ${remcalc(37)} 0;
+
+  *::selection,
+  ::selection {
+    background-color: #65ffcd;
```
Comment (jeysal) [inline src/components/Blog/Typography.js]: `neonGreen` from theme

File: src/components/Blog/Typography.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    `}
+
+    font-weight: 400;
+    color: ${props => props.theme.colors.text};
+    font-family: 'Roboto Mono', sans-serif;
+    padding: ${remcalc(4)};
+    background-color: #e9e9e9;
+
+    ::selection {
+      background-color: #65ffcd;
```
Comment (jeysal) [inline src/components/Blog/Typography.js]: ^

File: src/templates/blog.js [hunk 4]
```diff
@@ -58,30 +52,32 @@ const DisplayTitleCol = styled(Col)`
...
     }
   ])
+  const { numberOfPages, currentPage } = pageContext
+
+  const isFirst = currentPage === 1
+  const isLast = currentPage === numberOfPages
+  const prevPageNum =
+    currentPage - 1 === 1 ? `` : `page/${(currentPage - 1).toString()}`
+  const nextPageNum = (currentPage + 1).toString()
+  const prevPageLink = isFirst ? null : `/blog/${prevPageNum}`
```
Comment (jeysal) [inline src/templates/blog.js]: prev and next not symmetric here, which is confusing. can we unify

File: src/templates/blog.js [hunk 5]
```diff
@@ -58,30 +52,32 @@ const DisplayTitleCol = styled(Col)`
...
-  const mediumLink = 'https://medium.com/yld-blog'
-
   const breadcrumbData = generateBreadcrumbData(siteUrl, [
     {
       name: 'Blog',
       pathname: location.pathname,
       position: 2
     }
   ])
+  const { numberOfPages, currentPage } = pageContext
```
Comment (jeysal) [inline src/templates/blog.js]: extract into component with tests in follow up?

File: package.json [hunk 6]
```diff
@@ -4,6 +4,8 @@
   "version": "1.0.1",
   "author": "YLD <hello@yld.io>",
   "dependencies": {
+    "@mdx-js/mdx": "^1.5.1",
```
Comment (jeysal) [inline package.json]: can you check if we need all these?

Comment (omonk) [reply package.json]: Removed gatsby-remark-prismjs, everything else we need

**Pull request merged**